presto-spi/src/main/java/com/facebook/presto/spi/security/TokenAuthenticator.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+public interface TokenAuthenticator
+{
+    String getAuthenticationToken(String authenticationTokenRequest);
```

Comment:
What do you think about using `byte[]` or `ByteBuffer` or `Slice` instead of `String`? Binary representation is usually more versatile, as it allows to use binary encodings without re-encoding to Base64.

Reply:
Makes sense. I think the convention in Presto codebase is to use `Slice`, for example https://github.com/prestodb/presto/blob/8f9d9cd3a950898b185a4d67a5d95ebb44e2b6b7/presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java#L841

Reply:
Let's use `Slice` then